rust/lance/src/dataset/write/merge_insert.rs

```diff
@@ -606,6 +621,19 @@ impl MergeInsertJob {
     }
 }
 
+// Merger will store these statistics as it runs (for each batch)
```

Comment:
Since this is a public-facing struct, we should make public documentation comments (use `///` instead of `//`). Same goes for the fields.

```suggestion
/// Statistics about the completed merge operation.
```